src/main/java/Duke.java

```diff
@@ -1,10 +1,212 @@
+import java.util.Scanner;
+
 public class Duke {
+    protected static Task[] tasks = new Task[100];
+    protected static int numTasks = 0;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        greet();
+        acceptCommand();
+        exit();
+    }
+
+    public static void greet() {
+        printLine();
+        System.out.println("Hi there! I'm Domo the chat bot.\nWhat would you like to do?");
+    }
+
+    public static void exit() {
+        printLine();
+        System.out.println("Goodbye. Hope to see you again soon!");
+        printDomo();
+    }
+
+    public static void acceptCommand() {
+        String input = getInput();
+        while (!isBye(input)) {
+            String command = getCommand(input);
+            if (isValidCommand(command)) {
+                executeCommand(input);
+            } else {
+                System.out.println("Not a valid command. Please try again.");
+            }
+            input = getInput();
+        }
+    }
+
+    public static void executeCommand(String input) {
+        String command = getCommand(input);
+        System.out.println("\nCommand accepted!");
+        switch(command) {
+        case "list":
+            System.out.println("Here is your list so far:");
+            listTasks();
+            break;
+        case "mark":
+            executeMark();
+            break;
+        case "unmark":
+            executeUnmark();
+            break;
+        case "todo":
+            executeTodo(input);
+            break;
+        case "deadline":
+            executeDeadline(input);
+            break;
+        case "event":
+            executeEvent(input);
+            break;
+        }
+    }
```

Comment:
Perhaps you should think about adding a default case? Default case could possibly be used for catching invalid commands?

Reply:
Thank you for this comment, definitely helped improve the code's logic and avoided unnecessary work!